controller/src/controller.rs

```diff
@@ -271,27 +272,42 @@ impl<T: BottlerocketShadowClient> BrupopController<T> {
             let active_set_size = active_set.len();
             event!(Level::TRACE, ?active_set, "Found active set of nodes.");
 
-            // update time window: users can specify a update time window to operate Bottlerocket nodes update. If current time isn't within the time window,
-            // controller shouldn't have any action on it.
-            match within_time_window()? {
-                // If controller has already started a node update but the update time window stops,
-                // controller should respect it, continue to complete the update, and stop actions on remaining nodes.
-                // This logic will cooperate with 6 mins buffer strategy.
-                false => {
-                    // try to find out if any nodes are being updated. If yes, controller
-                    // will complete them and pause actions on other waitingForUpdate nodes.
-                    if !active_set.is_empty() {
-                        self.progress_active_set(active_set).await?;
-                    } else {
-                        sleep(ACTION_INTERVAL).await;
-                        continue;
-                    }
-                }
-                true => {
-                    if !active_set.is_empty() {
-                        self.progress_active_set(active_set).await?;
-                    }
+            // complete any ongoing updates even if we are outside of a scheduled window.
+            if !active_set.is_empty() {
+                self.progress_active_set(active_set).await?;
+            } else {
+                // calculate the duration between current time and next scheduled time, and then make
+                // controller sleep until next scheduled time point.
+                let next_scheduled_time = brupop_schedule
+                    .upcoming(Utc)
+                    .next()
+                    .context(controllerclient_error::GetScheduledDatetimeSnafu)?;
+                let duration_until_next_schedule = next_scheduled_time - Utc::now();
+
+                // covert chrono duration to std duration
+                let duration_until_next_schedule_std = duration_until_next_schedule
+                    .to_std()
+                    .context(controllerclient_error::ConvertToStdDurationSnafu)?;
+                event!(
+                    Level::INFO,
+                    next_schedule_time = ?next_scheduled_time,
+                    sleep = duration_until_next_schedule_std.as_secs(),
+                    "Began sleeping until next scheduled time point."
+                );
+                tokio::time::sleep(duration_until_next_schedule_std).await;
+            }
 
+            // cron expression is a time-based job scheduler user to run a specified job periodically at fixed times or periods of time (time window).
+            // fixed times: 0 10 * * * */At 10:00 on Monday.
```

Comment:
As written, this cron expression is: `At 10 minutes past the hour every hour` since `0` is the seconds, 10 is the minutes, and the next `*` is the hours.

For it to be `Every Monday at 10`, [from how I understand the `cron` library](https://github.com/zslayton/cron), I believe this would need to become: `0 0 10 * * Mon *`

Reply:
Yeah! You are correct! It should be 
```
// fixed times: 0 0 10 * * Mon */At 10:00 on every Monday.
```